### DISCUSSION TITLE
CONNAISSEUR 2.0 released!

### DISCUSSION THREAD

**Discussion by xopham** [sse-secure-systems/connaisseur#187] [Announcements]
## Connaisseur 2.0 is available now :tada:

As [announced end of April](https://github.com/sse-secure-systems/connaisseur/discussions/151) and after several months of hard work, Connaisseur 2.0 is finally here! Thanks to the great feedback from the community, we were able to tailor Connaisseur better to the needs of a modern container image signature infrastructure.

Based on the comments, issues, GitHub Discussions and other communication, we re-designed the configuration and modularized Connaisseur to make it more flexible and performant and for example include different signing solutions, multiple keys and registries. The main features of this release are:
- **multi-key support** :key::old_key::key:
- **multi-validator support** :detective::police_officer::female_detective::policeman:  
- **overall better usability** :hammer_and_wrench: 
- **improved Cosign support** (incl. auth. for private registries) :writing_hand: 
- **all new documentation** :memo: 

**Our [new documentation](https://sse-secure-systems.github.io/connaisseur/) is the best way to get started using version 2.0 :rocket:** 

Major changes are associated with:
- [ADR-3: Multi-Notary Configuration](https://sse-secure-systems.github.io/connaisseur/latest/adr/ADR-3_multi_notary_config/)
- [ADR-4: Modular Validation](https://sse-secure-systems.github.io/connaisseur/latest/adr/ADR-4_modular/)

You can find more details and all new features in the [release notes](https://github.com/sse-secure-systems/connaisseur/releases/tag/v2.0.0). Two exciting ones are:
- [added get_root_key utility](https://github.com/sse-secure-systems/connaisseur/pull/90/commits/3c6eb730585c61d05e97ee64c3d63619632843a4): *a utility to extract the public root key from Notary trust data of any DCT-signed image*
- [new ignore/validate label namespacebased validation](https://github.com/sse-secure-systems/connaisseur/pull/164/commits/ac856dc4a65f679af8ba4ef9cd99f1953a9da8a8): *much more flexible namespaced validation support*

**We really appreciate input from the community, so please keep sharing your feedback and ideas on the new version either as a comment below, a [GitHub Discussion](https://github.com/sse-secure-systems/connaisseur/discussions), an [issue](https://github.com/sse-secure-systems/connaisseur/issues) or contribute directly via a PR :pray:** 

:exclamation:**Note**: This is a major release as the user-facing configuration had to be adjusted which makes this a breaking change. Users that have using versions 1.x.x in a cluster and want to switch to version 2.0.0 now, should delete *CustomResourceDefinition* of Connaisseur that is used to validate the configuration and not typically deleted when uninstalling. This can be achieved via `kubectl delete customresourcedefinitions imagepolicies.connaisseur.policy`.

## What's up next? :next_track_button: 

With the next releases, we aim to focus on _performance, patching and stability improvements_ like increasing parallelization, implementing a more performant server and load tests. Furthermore, as Cosign is bound for its 1.0 release, we will extend integration into Connaisseur exposing some of the more advanced features.

**Comment by xopham** [comment]
@devopstales this release also fixes the shortcomings you mentioned in your [write-up](https://devopstales.github.io/home/k8s-connaisseur/). Thanks for the feedback :pray: much appreciated!

**Reply by apopaa** [reply]
Is there an equivalent in 2.0 of the 1.5 "make annihilate"? I'm trying to uninstall 2.0 but it is failing:
_helm uninstall connaisseur --namespace connaisseur
Error: warning: Hook pre-delete connaisseur/templates/webhook/job.yaml failed: Internal error occurred: failed calling webhook "connaisseur-svc.connaisseur.svc": Post https://connaisseur-svc.connaisseur.svc:443/mutate?timeout=30s: context deadline exceeded
bash-5.1#
bash-5.1#
bash-5.1# helm uninstall connaisseur --namespace connaisseur
Error: warning: Hook pre-delete connaisseur/templates/webhook/role.yaml failed: clusterroles.rbac.authorization.k8s.io "connaisseur-hook-role" already exists_

UPDATE:
I ended up doing this, I hope it's the right thing; would be nice to have one command that does it all:
_helm uninstall connaisseur --namespace connaisseur
kubectl delete all,mutatingwebhookconfigurations,clusterroles,clusterrolebindings,configmaps,imagepolicies,secrets,serviceaccounts,crds -lapp.kubernetes.io/instance=connaisseur
kubectl delete ns connaisseur_

**Reply by xopham** [reply]
Yes that should work. Actually the Makefile and thus `make annihilate` is still there. There is a seperate section in the docs: https://sse-secure-systems.github.io/connaisseur/v2.0.0/basics/#makefile

And we did extend the `helm uninstall instructions`: https://sse-secure-systems.github.io/connaisseur/v2.0.0/basics/#delete

Both installation and uninstallation can be a bit tricky bc the order is so critical as one otherwise ends in blocked state. We hope that improved a bit :pray:

In day-to-day situations actually the simple `helm uninstall` should suffice and we improved it somewhat. Also we'll be looking into upgrading as that is still an open flank. It's capture in #181

**Reply by devopstales** [reply]
@xopham Thanks for the update. I will test this new version and update my post.

**Reply by devopstales** [reply]
@xopham The new version is awesome. [I updated the post.](https://devopstales.github.io/kubernetes/k8s-connaisseur-v2/)

**Reply by xopham** [reply]
@devopstales thanks. Awesome blog post :slightly_smiling_face:
